.github/actions/setup-miniconda/action.yml

```diff
@@ -0,0 +1,53 @@
+name: Set up conda environment for testing
+
+description: Clean workspace and check out PyTorch
+
+inputs:
+  python-version:
+    description: If set to any value, don't use sudo to clean the workspace
+    required: false
+    type: string
+    default: "3.9"
+
+runs:
+  using: composite
+  steps:
+      - name: Install miniconda onto the machine
+        shell: bash -l {0}
+        run: |
+          MINICONDA_INSTALL_PATH_MACOS="${RUNNER_TEMP}/miniconda"
+          mkdir -p "${MINICONDA_INSTALL_PATH_MACOS}"
+          case ${RUNNER_OS}-${RUNNER_ARCH} in
+            macOS-ARM64)
+              MINICONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-arm64.sh"
+              ;;
+            macOS-X64)
+              MINICONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh"
+              ;;
+            *)
+              echo "Platform ${RUNNER_OS}-${RUNNER_ARCH} currently unsupported using this action"
+              exit 1
+              ;;
+          esac
+          curl -fsSL "${MINICONDA_URL}" -o "${MINICONDA_INSTALL_PATH_MACOS}/miniconda.sh"
+          bash "${MINICONDA_INSTALL_PATH_MACOS}/miniconda.sh" -b -u -p "${MINICONDA_INSTALL_PATH_MACOS}"
+          rm -rf "${MINICONDA_INSTALL_PATH_MACOS}/miniconda.sh"
+          echo "${MINICONDA_INSTALL_PATH_MACOS}/bin" >> $GITHUB_PATH
+      - name: Setup conda environment
+        shell: bash
+        env:
+          PYTHON_VERSION: ${{ inputs.python-version }}
+        run: |
+          CONDA_ENV="${RUNNER_TEMP}/conda_environment_${GITHUB_RUN_ID}"
+          conda create \
+            --yes \
+            --prefix "${CONDA_ENV}" \
+            "python=${PYTHON_VERSION}" \
+            cmake=3.22 \
+            conda-build=3.21 \
+            ninja=1.10 \
+            numpy=1.23 \
```

Comment:
1.23 is not available for Python-3.7 (and also, would break our OSS CI for example)
Perhaps pinned versions should be a configurable inputs for the action?

Reply:
I'll add a feature to pass an env file so you can install from there as well. In general though I do think it's valuable to have a pinned version of particular dependencies (like cmake) at least in initial install so we can enforce good behavior